stdlib/public/core/StringBridge.swift

```diff
@@ -141,43 +51,41 @@ extension String {
   @inline(never) @_semantics("stdlib_binary_only") // Hide the CF dependency
   public // SPI(Foundation)
   init(_cocoaString: AnyObject) {
+    // If the NSString is actually a Swift String in disguise, 
+    // we can just copy out the internal repr.
     if let wrapped = _cocoaString as? _NSContiguousString {
       self._core = wrapped._core
       return
     }
 
-    // "copy" it into a value to be sure nobody will modify behind
-    // our backs.  In practice, when value is already immutable, this
-    // just does a retain.
-    let cfImmutableValue
-      = _stdlib_binary_CFStringCreateCopy(_cocoaString) as AnyObject
-
-    let length = _swift_stdlib_CFStringGetLength(cfImmutableValue)
+    let length = _swift_stdlib_CFStringGetLength(_cocoaString)
 
     // Look first for null-terminated ASCII
-    // Note: the code in clownfish appears to guarantee
-    // nul-termination, but I'm waiting for an answer from Chris Kane
-    // about whether we can count on it for all time or not.
+    // FIXME: we're assuming "null-terminated ASCII" is a synonym for "ASCII". 
+    // This appears to be true, but we should verify that ObjC actually 
+    // guarantees this. Only a perf concern; soundness doesn't rely on this.
     let nulTerminatedASCII = _swift_stdlib_CFStringGetCStringPtr(
-      cfImmutableValue, kCFStringEncodingASCII)
+      _cocoaString, kCFStringEncodingASCII)
```

Comment:
This should really be using the SPIs from Foundation instead of doing the CF call (which is slower).

```
@interface NSString (SPI)
- (nullable const char *)_fastCStringContents:(BOOL)nullTerminationRequired;
@end
```

That will return the underlying C string pointer if it can be pulled out of the string.